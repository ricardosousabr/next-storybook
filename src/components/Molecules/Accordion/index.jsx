import Button from "../../Atoms/Button";
import { useState } from "react";

export default function Accordion({children}){
  const [open, setOpen] = useState(false);
  return (
    <>
      <div>
        <div>
          <p>List</p>
          <Button padding={"10px 20px"} borderRadius={"10px"} background={"aqua"} onClick={() => {setOpen(!open)}}>{open ? "-" : "+"}</Button>
        </div>
        <div>
          { open &&
            <ul>
              <li>{children}</li>
            </ul>
          }
        </div>
      </div>
    </>
  )
}
