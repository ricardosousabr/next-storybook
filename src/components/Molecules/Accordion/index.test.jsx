import renderer from "react-test-renderer";
import Accordion from "./index";

test("render accordion", () => {
  const tree = renderer.create(<Accordion />).toJSON();
  expect(tree).toMatchSnapshot();
});
