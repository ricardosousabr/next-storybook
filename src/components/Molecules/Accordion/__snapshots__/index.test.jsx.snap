// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render accordion 1`] = `
<div>
  <div>
    <p>
      List
    </p>
    <button
      background="aqua"
      borderRadius="10px"
      className="styles__ButtonStyle-sc-1hhwn61-0 iRCioL"
      onClick={[Function]}
      padding="10px 20px"
    >
      +
    </button>
  </div>
  <div />
</div>
`;
